routes/api/reviews.js

```diff
@@ -1,8 +1,166 @@
 const express = require("express");
 const router = express.Router();
 
-router.get("/test", (req, res) =>
-  res.json({ msg: "This is the reviews route" })
-);
+const Review = require("../../models/Review");
+const ListVersion = require("../../models/ListVersion");
+const TextString = require("../../models/TextString");
+
+// Show
+router.get("/:id", (req, res) => {
+  Review.findById(req.params.id)
+    .populate("description")
+    .populate({
+      path: "businessId",
+      populate: {
+        path: "displayName",
+        path: "description",
+      },
+    })
+    .populate({
+      path: "dishId",
+      populate: {
+        path: "name",
+        path: "description",
+      },
+    })
+    .populate({
+      path: "versionId",
+      populate: {
+        path: "listId",
+        populate: {
+          path: "name",
+          path: "description",
+        },
+      },
+    })
+    .then((review) => {
+      if (review) {
+        res.json(review);
+      } else {
+        res.status(404).json("No Such Review");
+      }
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Index
+router.get("/", (req, res) => {
+  Review.find({})
+    .populate("description")
+    .populate({
+      path: "businessId",
+      populate: {
+        path: "displayName",
+        path: "description",
+      },
+    })
+    .populate({
+      path: "dishId",
+      populate: {
+        path: "name",
+        path: "description",
+      },
+    })
+    .populate({
+      path: "versionId",
+      populate: {
+        path: "listId",
+        populate: {
+          path: "name",
+          path: "description",
+        },
+      },
+    })
+    .then((reviews) => {
+      res.json(reviews);
+    })
+    .catch((err) => res.status(404).json(err));
+});
+
+// Create
+router.post("/", async (req, res) => {
+  let { versionId, description, rating, businessId, dishId } = req.body;
+  description = await TextString.create(description)
+  .catch(err => console.log(err))
+
+  let review = new Review({
+    versionId,
+    description,
+    rating,
+    businessId,
+    dishId,
+  });
+
+  Review.create(review)
+    .then(async (review) => {
+      let listVersion = await ListVersion.findOne({ _id: review.versionId })
+      listVersion.reviews.push(review);
+      listVersion.save()
+        .catch(err => console.log(err))
+      res.json(review)
+    })
+    .catch((err) => res.json(err));
+});
+
+// Update
+router.put("/:id", async function (req, res) {
+  let {
+    versionId,
+    description,
+    rating,
+    businessId,
+    dishId,
+    overratedCount,
+    underratedCount,
```

Comment:
I would not put `overratedCount` and `underratedCount` here - let's make a separate endpoint to update them only, since they're the only fields who can get updated based on a user action (and not an admin).